spec/support/repository_factory.rb

```diff
@@ -0,0 +1,91 @@
+# frozen_string_literal: true
+
+require_relative "vcr"
+
+class StubRepository
+  attr_reader :full_name, :branches
+
+  def repo_file(repo_name, path)
+    File.read(Rails.root.join("spec/fixtures/repos/#{repo_name}", path.to_s))
+  end
+
+  def repo_json(repo_name, path)
```

Comment:
Is there a more descriptive name we can give this? I have a general idea what it does, but from the method name it's not clear.